app/services/preserved_object_handler.rb

```diff
@@ -60,7 +60,66 @@ def create
   end
 
   def check_existence
-    # this is just a stub
+    if invalid?
+      handler_results.add_result(PreservedObjectHandlerResults::INVALID_ARGUMENTS, errors.full_messages)
+    elsif PreservedObject.exists?(druid: druid)
+      Rails.logger.debug "check_existence #{druid} called"
+      if endpoint.endpoint_type.endpoint_class == 'online'
+        transaction_ok = with_active_record_transaction_and_rescue do
+          pres_object = PreservedObject.find_by!(druid: druid)
+          # FIXME: what if there is more than one associated pres_copy?
+          pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint) if pres_object
+
+          if pres_copy.version != pres_object.current_version
+            handler_results.add_result(PreservedObjectHandlerResults::PC_PO_VERSION_MISMATCH, pres_copy.class.name)
```

Comment:
added info to #403  to indicate the actual versions in this message (and maybe rollback message) -- that should happen both here and in confirm_version (#403 was created stemming from that PR review)